app/contacts/contact_tcp.py

```diff
@@ -73,18 +77,19 @@ async def accept(self, reader, writer):
         agent, _ = await self.services.get('contact_svc').handle_heartbeat(**profile)
         new_session = Session(id=self.generate_number(size=6), paw=agent.paw, connection=connection)
         self.sessions.append(new_session)
-        await self.send(new_session.id, agent.paw)
+        await self.send(new_session.id, agent.paw, timeout=5)
 
-    async def send(self, session_id, cmd):
+    async def send(self, session_id, cmd, timeout=60):
```

Comment:
We should default `timeout` to `None` here, as it is defaulted to `60` in `_attempt_connection` and this method just passes it through.  It also defaults to 60 in the instruction init/constructor.

Reply:
How about keeping the default here, but removing defaults in `_attempt_connection`?

Something like:
```
async def send(self, session_id, cmd, timeout=60):

async def _attempt_connection(self, session_id, connection, timeout):
```

Reply:
Sounds like a good change to have `_attempt_connection` have `timeout` as a required argument to the function and leave `send` with the current changes.

Reply:
yea, that makes sense and I can get behind it, default it in the public function.

Reply:
Okay, I'll proceed in that direction then.